File: Doodream/문자열 압축.js
function solution(s) {
  let compressedStringLengthArr = [];

  // i는 압축문자 단위가 s길이의 절반을 넘을수 없으므로 그만큼 반복한다.
  for (let i = 0; i < s.length / 2; i++) {
    let compressedStringLength = i + 1;
    let repeatStringCount = 1;
    let newString = "";

    // j는 압축문자가 다음문자와 같은지 s에 대해서 전부 비교해 나가본다.
    for (let j = 0; j < s.length; j += compressedStringLength) {
      let currentString = s.substring(j, j + compressedStringLength);
      let nextString = s.substring(
        j + compressedStringLength,
        j + 2 * compressedStringLength
      );

      //현재문자와 다음문자가 같다면 반복횟수 증가
      if (currentString === nextString) {
        repeatStringCount += 1;
        // 같지 않다면 더이상 반복이 안되는 것이므로 새로운 문자열에 반복횟숫를 넣어서 반복문자열을 넣는다.
      } else {
        if (repeatStringCount !== 1) {
          newString = newString + repeatStringCount + currentString;
          // 반복이 안된다면 현재 문자열이 나머지 문자열이 되므로 기존 문자열에 붙여 넣는 것이다.
        } else {
          newString = newString + currentString;
        }
        //반복 횟수 초기화
        repeatStringCount = 1;
      }
    }
    // 압축완료된 문자열 케이스 길이 하나 추가
    compressedStringLengthArr.push(newString.length);
  }

  var answer = Math.min(...compressedStringLengthArr);
  return answer;
}

console.log(solution("aabbaccc"));

File: KwangHo/문자열 압축.py
def solution(s):
    n = len(s) #문자열의 길이
    answer = n #최대값을 넣어준다
    
    for i in range(1,int(n/2+1)): #자를떄 1부터 ~ 절반까지
      dap = '' #임시값 초기화
      temp = s[0:i] #처음 자를값 초기화
      cnt = 1 #최소1개는 있으니 1로
      for j in range(i,n,i): #2개씩 자르면 2~끝까지-1, 2개씩 증가
        if temp == s[j:j+i]: #자른값이랑 나머지값들 비교
          cnt+=1 #같다면 +1
        else : #아니라면
          if cnt>1: dap+=str(cnt) #2개 이상이라면 같은값이 존재니깐 숫자추가
          dap += temp #자른값 추가
          temp = s[j:j+i] #같은 같이 없으니 다음값으로 자른값 갱신
          cnt = 1 # 최소1 초기화

      if cnt>1: dap+=str(cnt) #위와같음
      dap += temp #위와같음 => 한번 더써주는이유는 마지막 남은값 때문

      answer = min(answer,len(dap)) #작은값 찾기
    return answer

'''
처음 문제를 읽고 아하 이렇게 풀면되겟다.
했는데 머릿속과 의사코드로는 그려졌는데
막상 코드로 옮길려니깐 생각보다 잘 그려지지 않았고
파이썬 문법도 자꾸 틀려서 조금 해맸다.ㅠ
'''